### PR TITLE
Divyanshu | Stage | Convert Setting side bar all tabs into material

### PR DESCRIPTION
**Pull request by nisha-walkover**
https://app.clickup.com/t/85ztgavyt

### CODE REVIEW

File: apps/web-giddh/src/app/settings/branch/branch.component.html [hunk 1]
```diff
@@ -287,17 +287,11 @@ <h3>{{ localeData?.create_new_branch }}</h3>
...
-            bsModal
-            #confirmationModal="bs-modal"
-            [config]="{ backdrop: 'static', keyboard: 'true' }"
-            tabindex="-1"
-            class="modal fade"
-            role="dialog"
-        >
+        <ng-template #confirmationModal2>
+            Hello
             <div class="modal-dialog modal-md">
```
Comment (nisha-walkover) [inline apps/web-giddh/src/app/settings/branch/branch.component.html:198]: data is not coming in dialog

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.html [hunk 2]
```diff
@@ -61,17 +61,23 @@ <h3 class="font-20 bold mb-2">{{localeData?.sms?.form_heading}}</h3>
...
                         <ng-container *ngIf="!(isGmailIntegrated$ | async)">
-                            <a *ngIf="!isElectron" class="btn btn-primary" [href]="gmailAuthCodeUrl$ | async">{{
-                                localeData?.email?.connect_gmail
-                            }}</a>
+                            <button
+                                *ngIf="!isElectron"
+                                mat-stroked-button
+                                color="primary"
+                                [href]="gmailAuthCodeUrl$ | async"
+                            >
```
Comment (nisha-walkover) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.html:70]: please check href on button

File: apps/web-giddh/src/app/permissions/components/details/permission.details.component.ts [hunk 3]
```diff
@@ -138,14 +160,15 @@ export class PermissionDetailsComponent implements OnInit, OnDestroy {
     public getScopeDataReadyForAPI(data): Scope[] {
         let arr: Scope[];
         arr = forEach(data?.scopes, (page: Scope) => {
-            remove(page.permissions, (o: Permission) => !o.isSelected);
+            remove(page.permissions, (o: Permission) => !o.isSelected || o.code === 'SELECT-ALL');
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/permissions/components/details/permission.details.component.ts]: do not use single characted

File: apps/web-giddh/src/app/permissions/components/details/permission.details.component.ts [hunk 4]
```diff
@@ -165,12 +188,29 @@ export class PermissionDetailsComponent implements OnInit, OnDestroy {
...
         } else {
             // copy role scenario
-            return this.generateUIFromExistedRole();
+            response = this.generateUIFromExistedRole();;
+        }
+
+        if (response) {
+            response.forEach(item => {
+                let count = 0;
+                item.permissions.forEach(item => {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/permissions/components/details/permission.details.component.ts]:
```suggestion
                item?.permissions?.forEach(item => {
```

File: apps/web-giddh/src/app/permissions/components/details/permission.details.component.ts [hunk 5]
```diff
@@ -291,23 +331,27 @@ export class PermissionDetailsComponent implements OnInit, OnDestroy {
     }
 
     public toggleItems(pageName: string, event: any) {
+
         let res = find(this.roleObj?.scopes, (o: Scope) => o.name === pageName);
         if (res) {
-            map(res.permissions, (o: Permission) => o.isSelected = event.target?.checked ? true : false);
+            map(res.permissions, (o: Permission) => o.isSelected = event.checked ? true : false);
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/permissions/components/details/permission.details.component.ts]: do not use single character

File: apps/web-giddh/src/app/permissions/components/details/permission.details.component.ts [hunk 6]
```diff
@@ -291,23 +331,27 @@ export class PermissionDetailsComponent implements OnInit, OnDestroy {
...
         }
     }
 
     public toggleItem(pageName: string, item: Permission, event: any) {
         let res = find(this.roleObj?.scopes, (o: Scope) => o.name === pageName);
-        if (event.target?.checked) {
-            let idx = findIndex(res.permissions, (o: Permission) => o.isSelected === false);
+        if (event.checked) {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/permissions/components/details/permission.details.component.ts:343]:
```suggestion
        if (event?.checked) {
```

File: apps/web-giddh/src/app/permissions/components/details/permission.details.component.ts [hunk 7]
```diff
@@ -291,23 +331,27 @@ export class PermissionDetailsComponent implements OnInit, OnDestroy {
...
+            let idx = findIndex(res.permissions, (o: Permission) => {
+                return o.isSelected === false && o.code !== 'SELECT-ALL';
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/permissions/components/details/permission.details.component.ts]: do not use single character

File: apps/web-giddh/src/app/permissions/components/details/permission.details.html [hunk 8]
```diff
@@ -8,16 +8,22 @@
...
             <span class="lead">{{ newRole.name }}</span>
-            <button class="btn btn-md btn-primary pull-left" (click)="goToRoles()">
-                {{commonLocaleData?.app_back}}
-            </button>
-            <button *ngIf="!newRole.isUpdateCase" class="btn btn-md btn-success pull-right mr-0" (click)="addNewRole()">
+            <button mat-stroked-button class="pull-left" (click)="goToRoles()">{{commonLocaleData?.app_back}}</button>
+            <button
+                mat-stroked-button
+                color="primary"
+                *ngIf="!newRole.isUpdateCase"
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/permissions/components/details/permission.details.html]:
```suggestion
                *ngIf="!newRole?.isUpdateCase"
```

File: apps/web-giddh/src/app/permissions/components/details/permission.details.html [hunk 9]
```diff
@@ -27,103 +33,87 @@
...
-                                [disabled]="checkForAlreadyExistInPageArray(page)"
-                                *ngFor="let page of pageList;"
-                                [ngValue]="page"
-                            >
-                                {{page}}
-                            </option>
-                        </select>
+                        <dropdown-field
+                            [placeholder]="localeData?.option_select_pages"
+                            [readonly]="newRole.isFixed"
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/permissions/components/details/permission.details.html]:
```suggestion
                            [readonly]="newRole?.isFixed"
```

File: apps/web-giddh/src/app/permissions/components/details/permission.details.html [hunk 10]
```diff
@@ -27,103 +33,87 @@
...
-                                                    aria-hidden="true"
-                                                ></i>
-                                            </td>
-                                        </ng-container>
-                                    </tr>
-                                </tbody>
-                            </table>
-                        </div>
+                    <div>
+                        <h2 class="pd-l05">{{ page.name }}</h2>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/permissions/components/details/permission.details.html]:
```suggestion
                        <h2 class="pd-l05">{{ page?.name }}</h2>
```

File: apps/web-giddh/src/app/permissions/components/details/permission.details.html [hunk 11]
```diff
@@ -27,103 +33,87 @@
...
+                        <a
+                            class="close pull-right"
+                            href="javascript:void(0);"
+                            (click)="removePageFromScope(page.name)"
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/permissions/components/details/permission.details.html]:
```suggestion
                            (click)="removePageFromScope(page?.name)"
```

File: apps/web-giddh/src/app/permissions/components/details/permission.details.html [hunk 12]
```diff
@@ -27,103 +33,87 @@
...
+                            *ngIf="!newRole.isFixed"
+                            >×</a
+                        >
                     </div>
+                    <table mat-table [dataSource]="page.permissions" class="w-100 mr-b4">
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/permissions/components/details/permission.details.html]:
```suggestion
                    <table mat-table [dataSource]="page?.permissions" class="w-100 mr-b4">
```

File: apps/web-giddh/src/app/permissions/components/details/permission.details.html [hunk 13]
```diff
@@ -27,103 +33,87 @@
...
+                        <ng-container matColumnDef="admin">
+                            <th mat-header-cell *matHeaderCellDef>{{ newRole.name }}</th>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/permissions/components/details/permission.details.html]:
```suggestion
                            <th mat-header-cell *matHeaderCellDef>{{ newRole?.name }}</th>
```

```

File: apps/web-giddh/src/app/permissions/components/details/permission.details.html [hunk 14]
```diff
@@ -27,103 +33,87 @@
...
+                            <td mat-cell *matCellDef="let element, let i = index">
+                                <ng-container *ngIf="element.code === 'SELECT-ALL'">
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/permissions/components/details/permission.details.html]:
```suggestion
                                <ng-container *ngIf="element?.code === 'SELECT-ALL'">
```
check element and page in overall file

File: apps/web-giddh/src/app/permissions/components/list/permission-list.html [hunk 15]
```diff
@@ -6,26 +6,42 @@
...
             <div class="row mr-0 ml-0">
                 <div class="col-md-3 col-sm-3 col-12 p-0 role-head cp">
                     <span class="vertical-center">
                         <h1>{{role.name}}</h1>
-                        <div class="text-center mr-t2 on-hover">
-                            <button *ngIf="!role.isFixed" class="btn btn-sm btn-primary" (click)="updateRole(role);">
+                        <div class="text-center mr-t2 on-hover d-flex column-gap1">
+                            <button
+                                mat-stroked-button
+                                *ngIf="!role.isFixed"
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/permissions/components/list/permission-list.html:30]:
```suggestion
                                *ngIf="!role?.isFixed"
```
change in over all file

File: apps/web-giddh/src/app/permissions/components/list/permission.list.component.ts [hunk 16]
```diff
@@ -104,9 +111,14 @@ export class PermissionListComponent implements OnInit, AfterViewInit, OnDestroy
...
 
+    /**
+     * Close Permission Dialog
+     *
+     * @memberof PermissionListComponent
+     */
     public closePopupEvent(userAction) {
         this.showPopup = false;
-        this.permissionModel.hide();
+        this.permissionDialogRef.close();
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/permissions/components/list/permission.list.component.ts:121]:
```suggestion
        this.permissionDialogRef?.close();
```

```

File: apps/web-giddh/src/app/permissions/components/model/permission.model.component.ts [hunk 17]
```diff
@@ -101,12 +115,17 @@ export class PermissionModelComponent implements OnInit, OnDestroy {
     }
 
     public selectAllPages(event) {
-        if (event.target?.checked) {
+        if (event.checked) {
+            this.selectedValues = [];
+            this.newRoleObj.pageList.forEach((item: IPage) => {
+                item.isSelected = true;
+                this.selectedValues.push(item)
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/permissions/components/model/permission.model.component.ts]:
```suggestion
                this.selectedValues.push(item);
```

File: apps/web-giddh/src/app/settings/address-settings/address-settings.component.html [hunk 18]
```diff
@@ -27,212 +27,207 @@
...
+                                        }
+                                    "
+                                >
+                                </ng-container>
+                            </div>
+                        </th>
+                        <td mat-cell *matCellDef="let element">
+                            <div>
+                                <ng-container *ngIf="element.taxType && element.taxNumber">
+                                    {{ element.taxNumber }}
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/address-settings/address-settings.component.html]:
```suggestion
                                    {{ element?.taxNumber }}

change in overall element in file

```

File: apps/web-giddh/src/app/settings/address-settings/address-settings.component.ts [hunk 19]
```diff
@@ -260,26 +268,34 @@ export class AddressSettingsComponent implements OnInit, OnChanges, OnDestroy {
...
+            height: '100vh !important',
+            disableClose: true,
+            position: {
+                right: '0',
+                top: '0'
+            }
+        });
+    }
+
+    public closeAccountAsidePane(): void{
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/address-settings/address-settings.component.ts]: make document for this

File: apps/web-giddh/src/app/settings/address-settings/address-settings.component.ts [hunk 20]
```diff
@@ -337,7 +353,7 @@ export class AddressSettingsComponent implements OnInit, OnChanges, OnDestroy {
     public handleUpdateAddress(address: any): void {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/address-settings/address-settings.component.ts:358]: make document for this

File: apps/web-giddh/src/app/settings/address-settings/address-settings.component.ts [hunk 21]
```diff
@@ -466,10 +482,18 @@ export class AddressSettingsComponent implements OnInit, OnChanges, OnDestroy {
      * @param {*} address Selected address
      * @memberof AddressSettingsComponent
      */
-    public showConfirmationModal(address: any): void {
+    // public showConfirmationModal(address: any): void {
+    //     this.selectedAddress = address;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/address-settings/address-settings.component.ts]: remove commented code

File: apps/web-giddh/src/app/settings/address-settings/address-settings.component.ts [hunk 22]
```diff
@@ -466,10 +482,18 @@ export class AddressSettingsComponent implements OnInit, OnChanges, OnDestroy {
...
+    //     this.deleteAddressConfirmationModal?.show();
+    // }
+
+    public showConfirmationModal(address: any) {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/address-settings/address-settings.component.ts:490]: make doucument for all methods where you were changes in over all file

File: apps/web-giddh/src/app/settings/address-settings/address-settings.component.ts [hunk 23]
```diff
@@ -466,10 +482,18 @@ export class AddressSettingsComponent implements OnInit, OnChanges, OnDestroy {
...
         this.selectedAddress = address;
-        this.deleteAddressConfirmationModal?.show();
-    }
+       this.deleteAddressConfirmationModalRef =  this.dialog.open(this.deleteAddressConfirmationModal, {
+            panelClass: 'modal-dialog',
+            width: '1000px',
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/address-settings/address-settings.component.ts]:
```suggestion
            width: '1000px'
```

File: apps/web-giddh/src/app/settings/address-settings/address-settings.component.ts [hunk 24]
```diff
@@ -482,17 +506,14 @@ export class AddressSettingsComponent implements OnInit, OnChanges, OnDestroy {
...
-        this.deleteAddressConfirmationModal.hide();
+        this.deleteAddressConfirmationModalRef.close();
     }
 
     /**
      * Handles the cancel operation
      *
      * @memberof AddressSettingsComponent
      */
-    public onCancel(): void {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/address-settings/address-settings.component.ts:493]: remove complete code

File: apps/web-giddh/src/app/settings/branch/branch.component.ts [hunk 25]
```diff
@@ -115,13 +124,23 @@ export class BranchComponent implements OnInit, AfterViewInit, OnDestroy {
         private _generalService: GeneralService,
         private settingsUtilityService: SettingsUtilityService,
         private toasterService: ToasterService,
-        private settingsBranchService: SettingsBranchService
+        private settingsBranchService: SettingsBranchService,
+        public dialog: MatDialog,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/branch/branch.component.ts]:
```suggestion
        public dialog: MatDialog
```

File: apps/web-giddh/src/app/settings/branch/branch.component.ts [hunk 26]
```diff
@@ -212,7 +227,27 @@ export class BranchComponent implements OnInit, AfterViewInit, OnDestroy {
...
+
+    /**
+     * Open Create company dialog 
+     *
+     * @memberof BranchComponent
+     */
+    public openCreateCompanyDialog(): void {
+        this.dialog.open(this.addCompanyModal, {
+            panelClass: 'modal-dialog',
+            width: '1000px',
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/branch/branch.component.ts]:
```suggestion
            width: '1000px'
```

File: apps/web-giddh/src/app/settings/branch/branch.component.ts [hunk 27]
```diff
@@ -536,11 +568,13 @@ export class BranchComponent implements OnInit, AfterViewInit, OnDestroy {
         const unarchivedBranches = this.unFilteredBranchList?.filter(currentBranch => !currentBranch?.isArchived);
         if (unarchivedBranches?.length > 1 || branch?.isArchived) {
             this.branchStatusToUpdate = branch;
-            this.statusModal?.show();
+            this.statusDialogRef = this.dialog.open(this.statusDialog, {
+                panelClass: 'modal-dialog',
+                width: '1000px',
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/branch/branch.component.ts]:
```suggestion
                width: '1000px'
```

```

File: apps/web-giddh/src/app/settings/branch/branch.component.ts [hunk 28]
```diff
@@ -560,7 +594,7 @@ export class BranchComponent implements OnInit, AfterViewInit, OnDestroy {
             } else {
                 this.toasterService.errorToast(response?.message);
             }
-            this.statusModal?.hide();
+            this.statusDialogRef.close();
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/branch/branch.component.ts:597]:
```suggestion
            this.statusDialogRef?.close();
```

File: apps/web-giddh/src/app/settings/discount/discount.component.ts [hunk 29]
```diff
@@ -1,92 +1,107 @@
...
+    /** This will hold common JSON data */
     public commonLocaleData: any = {};
     /** True if get all discounts api call in progress */
     public isLoading: boolean = false;
+    /** Holds Mat Table Display columns */
+    public displayedColumns: string[] = ['number', 'name', 'action'];
+    /** Holds Discount Confirmation Dialog Ref */
+    public discountConfirmationDialogRef: MatDialogRef<any>;
+    /** Holds Create New Account Dialog Ref */
+    public createNewAccountDialogRef: MatDialogRef<any>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/discount/discount.component.ts]:
```suggestion
    public createNewAccountDialogRef: MatDialogRef<any>;
```

File: apps/web-giddh/src/app/settings/discount/discount.component.ts [hunk 30]
```diff
@@ -103,31 +118,56 @@ export class DiscountComponent implements OnInit, OnDestroy {
...
+     * Open delete discount confirmation dialog
+     *
+     * @param {string} uniqueName
+     * @memberof DiscountComponent
+     */
+    public showDeleteDiscountDialog(uniqueName: string): void {
         this.deleteRequest = uniqueName;
-        this.discountConfirmationModel?.show();
+        this.discountConfirmationDialogRef = this.dialog.open(this.discountConfirmationDialog, {
+            panelClass: 'modal-dialog',
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/discount/discount.component.ts]:
```suggestion
            panelClass: 'modal-dialog'
```

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.ts [hunk 31]
```diff
@@ -231,7 +247,7 @@ export class SettingIntegrationComponent implements OnInit, AfterViewInit {
                         this.razorPayObj.account = { name: null, uniqueName: null };
                         this.forceClearLinkAccount$ = observableOf({ status: true });
                     }
-                    this.razorPayObj.password = 'YOU_ARE_NOT_ALLOWED';
+                    this.razorPayObj.password = o?.razorPayForm?.userName ? 'YOU_ARE_NOT_ALLOWED' : '';
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.ts:250]: do not use single character

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.ts [hunk 32]
```diff
@@ -1170,17 +1192,17 @@ export class SettingIntegrationComponent implements OnInit, AfterViewInit {
      */
     public openCreateNewAccountUserModal(bankAccount: any): void {
         this.activeBankAccount = bankAccount;
-        this.createNewAccountUserModal?.show();
+        this.createNewAccountUserModalRef = this.dialog.open(this.createNewAccountUserModal, {
+            panelClass: 'modal-dialog',
+            width: '1000px',
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.ts]:
```suggestion
            width: '1000px'
```

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.ts [hunk 33]
```diff
@@ -1281,16 +1302,10 @@ export class SettingIntegrationComponent implements OnInit, AfterViewInit {
...
-    /**
-     * This will close the delete bank account login confirmation modal
-     *
-     * @memberof SettingIntegrationComponent
-     */
-    public closeDeleteBankAccountLoginConfirmationModal(): void {
-        this.confirmationModal?.hide();
+        this.confirmationModalRef = this.dialog.open(this.confirmationModal, {
+            panelClass: 'modal-dialog',
+            width: '1000px',
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.ts]:
```suggestion
            width: '1000px'
```

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.ts [hunk 34]
```diff
@@ -1303,16 +1318,10 @@ export class SettingIntegrationComponent implements OnInit, AfterViewInit {
...
-    /**
-     * This will close the edit account user modal
-     *
-     * @memberof SettingIntegrationComponent
-     */
-    public closeEditAccountUserModal(): void {
-        this.editAccountUserModal?.hide();
+        this.editAccountUserModalRef = this.dialog.open(this.editAccountUserModal, {
+            panelClass: 'modal-dialog',
+            width: '1000px',
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.ts]:
```suggestion
            width: '1000px'
```

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.ts [hunk 35]
```diff
@@ -1323,16 +1332,10 @@ export class SettingIntegrationComponent implements OnInit, AfterViewInit {
...
-    /**
-     * This will close the edit account modal
-     *
-     * @memberof SettingIntegrationComponent
-     */
-    public closeEditAccountModal(): void {
-        this.editAccountModal?.hide();
+        this.editAccountModalRef = this.dialog.open(this.editAccountModal, {
+            panelClass: 'modal-dialog',
+            width: '1000px',
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.ts]:
```suggestion
            width: '1000px'
```
```

File: apps/web-giddh/src/app/settings/profile/setting.profile.component.ts [hunk 36]
```diff
@@ -276,6 +283,9 @@ export class SettingProfileComponent implements OnInit, OnDestroy {
 
         this.route.params.pipe(takeUntil(this.destroyed$)).subscribe(params => {
             this.currentTab = (params['referrer']) ? params['referrer'] : "personal";
+            if ((params['referrer']) === 'personal') { this.activeTabIndex = 0; }
+            else if ((params['referrer']) === 'address') { this.activeTabIndex = 1; }
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/profile/setting.profile.component.ts]: format code

File: apps/web-giddh/src/app/settings/tags/tags.component.ts [hunk 37]
```diff
@@ -132,4 +153,49 @@ export class SettingsTagsComponent implements OnInit, OnDestroy {
...
+     * @memberof SettingsTagsComponent
+     */
+    public deleteTag(tag: TagInterface): void {
+        this.setTagValue(tag);
+        let message = this.localeData?.remove_tag;
+        message = message?.replace("[TAG_NAME]", tag.name);
+        this.confirmationMessage = message;
+        this.dialog.open(this.confirmationModal, {
+            panelClass: 'modal-dialog',
+            width: '1000px',
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/tags/tags.component.ts]:
```suggestion
            width: '1000px'
```

File: apps/web-giddh/src/app/settings/warehouse/warehouse.component.ts [hunk 38]
```diff
@@ -135,7 +124,8 @@ export class WarehouseComponent implements OnInit, OnDestroy, AfterViewInit {
         private settingsProfileService: SettingsProfileService,
         private toasterService: ToasterService,
         private warehouseActions: WarehouseActions,
-        private settingsWarehouseService: SettingsWarehouseService
+        private settingsWarehouseService: SettingsWarehouseService,
+        public dialog: MatDialog,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/warehouse/warehouse.component.ts]:
```suggestion
        public dialog: MatDialog
```

File: apps/web-giddh/src/app/settings/warehouse/warehouse.component.ts [hunk 39]
```diff
@@ -180,9 +170,12 @@ export class WarehouseComponent implements OnInit, OnDestroy, AfterViewInit {
      *
      * @memberof WarehouseComponent
      */
+
     public openCreateWarehouseModal(): void {
         this.startOnBoarding();
-        this.warehouseOnBoardingModal?.show();
+        this.dialog.open(this.warehouseOnBoardingModal, {
+            panelClass: 'modal-dialog',
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/warehouse/warehouse.component.ts]:
```suggestion
            panelClass: 'modal-dialog'
```

File: apps/web-giddh/src/app/settings/warehouse/warehouse.component.ts [hunk 40]
```diff
@@ -338,26 +339,9 @@ export class WarehouseComponent implements OnInit, OnDestroy, AfterViewInit {
...
-     *
-     * @memberof WarehouseComponent
-     */
-    public toggleBodyClass(): void {
-        if (this.asideEditWarehousePane === 'in') {
-            document.querySelector('body').classList.add('fixed');
-        } else {
-            document.querySelector('body').classList.remove('fixed');
-        }
+        this.asideAccountAsidePaneDialogRef.close()
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/warehouse/warehouse.component.ts]:
```suggestion
        this.asideAccountAsidePaneDialogRef.close();
```

File: apps/web-giddh/src/app/settings/warehouse/warehouse.component.ts [hunk 41]
```diff
@@ -573,7 +557,10 @@ export class WarehouseComponent implements OnInit, OnDestroy, AfterViewInit {
     public confirmStatusUpdate(warehouse: any): void {
         if (!warehouse?.isDefault || warehouse?.isArchived) {
             this.warehouseStatusToUpdate = warehouse;
-            this.statusModal?.show();
+            this.statusModalRef = this.dialog.open(this.statusModal, {
+                panelClass: 'modal-dialog',
+                width: '1000px',
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/warehouse/warehouse.component.ts]:
```suggestion
                width: '1000px'
```

File: apps/web-giddh/src/app/shared/aside-menu-account/aside.menu.account.component.ts [hunk 42]
```diff
@@ -153,12 +152,12 @@ export class AsideMenuAccountInContactComponent implements OnInit, OnDestroy {
...
 
-    public showDeleteAccountModal() {
+    public showDeleteAccountModal(): void {
         this.deleteAccountmodalRef = this.dialog.open(this.deleteAccountModal);
     }
 
-    public hideDeleteAccountModal() {
-        this.deleteAccountmodalRef?.close()
+    public hideDeleteAccountModal(): void {
+        this.deleteAccountmodalRef.close()
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/aside-menu-account/aside.menu.account.component.ts:160]:
```suggestion
        this.deleteAccountmodalRef?.close()
```

File: apps/web-giddh/src/app/shared/create-address/create-address.component.ts [hunk 43]
```diff
@@ -57,13 +55,17 @@ export class CreateAddressComponent implements OnInit, OnDestroy {
...
     /** List of entities which can be archived */
     public entityArchived: string[] = ["BRANCH", "WAREHOUSE"];
+    /** Holds Selected Entity */
+    public selectedEntity: any[] = [];
 
     constructor(
         private formBuilder: UntypedFormBuilder,
         private toasterService: ToasterService,
-        private pageLeaveUtilityService: PageLeaveUtilityService
+        private pageLeaveUtilityService: PageLeaveUtilityService,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/create-address/create-address.component.ts]:
```suggestion
        private pageLeaveUtilityService: PageLeaveUtilityService
```

File: apps/web-giddh/src/app/shared/create-address/create-address.component.ts [hunk 44]
```diff
@@ -73,94 +75,129 @@ export class CreateAddressComponent implements OnInit, OnDestroy {
...
+                    state: [{ value: this.addressToUpdate.stateCode, disabled: !!this.addressToUpdate.taxNumber && this.addressConfiguration.tax && this.addressConfiguration.tax.name === 'GSTIN' }, !this.addressConfiguration.countyList?.length ? Validators.required : null],
+                    stateLabel: [this.addressToUpdate?.stateName && this.addressToUpdate?.stateCode ? this.addressToUpdate?.stateCode + ' - ' + this.addressToUpdate?.stateName : ''],
+                    county: [this.addressToUpdate.county?.code, this.addressConfiguration.countyList?.length ? Validators.required : null],
+                    address: [this.addressToUpdate.address, this.addressToUpdate.taxNumber && this.addressConfiguration.tax && this.addressConfiguration.tax.name === 'GSTIN' ? [Validators.required] : []],
+                    linkedEntity: [this.addressConfiguration.linkedEntities?.filter((item) => {
+                        return item?.uniqueName ===
+                            this.addressToUpdate.linkedEntities?.filter(i => i?.uniqueName === item?.uniqueName)[0]?.uniqueName
+                    })],
+                    pincode: [this.addressToUpdate.pincode]
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/create-address/create-address.component.ts:134]:
```suggestion
                    pincode: [this.addressToUpdate.pincode];
```

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/permissions/components/details/permission.details.component.ts (hunk 3 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/permissions/components/details/permission.details.component.ts]: Done

File: apps/web-giddh/src/app/permissions/components/details/permission.details.component.ts (hunk 6 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/permissions/components/details/permission.details.component.ts:343]: It always present, it is in mat checkbox

File: apps/web-giddh/src/app/permissions/components/list/permission.list.component.ts (hunk 16 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/permissions/components/list/permission.list.component.ts:121]: No use

File: apps/web-giddh/src/app/settings/branch/branch.component.ts (hunk 28 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/settings/branch/branch.component.ts:597]: No use

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.ts (hunk 31 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/settings/integration/setting.integration.component.ts:250]: Old code

File: apps/web-giddh/src/app/settings/profile/setting.profile.component.ts (hunk 36 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/settings/profile/setting.profile.component.ts]: already formatted, but still format in clear way

File: apps/web-giddh/src/app/shared/aside-menu-account/aside.menu.account.component.ts (hunk 42 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/shared/aside-menu-account/aside.menu.account.component.ts:160]: no use

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**